scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -2008,14 +2029,14 @@ private void determineCipherSuitesFromConfig() {
 			// try to guess his intentions from properties he has set
 			List<CipherSuite> ciphers = new ArrayList<>();
 			boolean certificates = isConfiguredWithKeyPair() || config.trustCertificateTypes != null;
-
+			boolean recommendedOnly = Boolean.FALSE.equals(recommendedCipherSuitesOnly);
```

Comment:
I would prefer that we define a default value for `recommendedCipherSuitesOnly` before to call `determineCipherSuitesFromConfig` instead of this test here.

Reply:
If the value is initialized, the checks for explicit cipher suites selection will be either not that easy (currently the check is only for ! null), or the verification logic must be changed.

Reply:
Maybe I don't get your point, but I don't see any issue if we do somthing like this in `build()`: 

```java
if (config.supportedCipherSuites == null || config.supportedCipherSuites.isEmpty()) {
    if (recommendedCipherSuitesOnly == null){
        recommendedCipherSuitesOnly = true;
    }
    determineCipherSuitesFromConfig();
}
```

Reply:
OK, I was mislead ...